vector/src/main/java/im/vector/app/features/room/VectorRoomDisplayNameFallbackProvider.kt

```diff
@@ -29,17 +29,10 @@ class VectorRoomDisplayNameFallbackProvider(
     }
 
     override fun getNameForEmptyRoom(isDirect: Boolean, leftMemberNames: List<String>): String {
-        return if (leftMemberNames.isEmpty()) {
-            context.getString(R.string.room_displayname_empty_room)
+        return if (isDirect && leftMemberNames.isNotEmpty()) {
+            leftMemberNames.first()
         } else {
-            val was = when (val size = leftMemberNames.size) {
-                1    -> getNameFor1member(leftMemberNames[0])
-                2    -> getNameFor2members(leftMemberNames[0], leftMemberNames[1])
-                3    -> getNameFor3members(leftMemberNames[0], leftMemberNames[1], leftMemberNames[2])
-                4    -> getNameFor4members(leftMemberNames[0], leftMemberNames[1], leftMemberNames[2], leftMemberNames[3])
-                else -> getNameFor4membersAndMore(leftMemberNames[0], leftMemberNames[1], leftMemberNames[2], size - 3)
-            }
-            context.getString(R.string.room_displayname_empty_room_was, was)
+            context.getString(R.string.room_displayname_empty_room)
```

Comment:
We just want to override the element-android behaviour when a direct is left.
Please copy/paste the element-android code in your `else` block.